core/src/utils/prove.rs

```diff
@@ -164,15 +172,50 @@ where
             );
         }
     };
+    make_checkpoint_span.exit();
 
     // Commit to the shards.
     #[cfg(debug_assertions)]
     let mut debug_records: Vec<ExecutionRecord> = Vec::new();
+
+    let commit_span = tracing::debug_span!("Commit to shards").entered();
     let mut deferred = ExecutionRecord::new(program.clone().into());
     let mut state = public_values.reset();
     let nb_checkpoints = checkpoints.len();
     let mut challenger = prover.config().challenger();
     vk.observe_into(&mut challenger);
+
+    let (tx, rx) = sync_channel::<Vec<ExecutionRecord>>(
+        opts.shard_batch_size * opts.shard_chunking_multiplier,
```

Comment:
why is this opts.shard_chunking_multiplier here?

Reply:
I am using it for channel capacity. I could make a new field in Opts

Reply:
Maybe jsut rename that field, I don't think we use it anymore anyways. Or maybe add a new field